src/mpyl/run_plan.py

```diff
@@ -164,7 +142,7 @@ def write_to_pickle_file(self):
     def write_to_json_file(self):
         run_plan: dict = {}
 
-        for stage, executions in self._full_plan.items():
+        for stage, executions in self._selected_plan.items():
```

Comment:
Both `_full_plan` and `_selected_plan` were readily available, is the switch to `_selected_plan` purely because that will now either contain the full plan (in case no projects were selected) or a selected plan?

Reply:
Great catch! 

The `RunPlan` class keeps information about two things:
* `full_plan` → what was automatically discovered based on the changed files in the PR or tag. This is never modified. _(By the way I'm considering renaming this to `discovered_plan` to make it more clear)_
* `selected_plan` → manual modifications made by a developer to the `full_plan` (e.g. selecting a specific project to deploy)

The run plan should always operate on the _selected plan_, and never the full plan _(technically we do use the full plan for one thing during the deploy stage, but that's not relevant for now)_.

So the fact that this was using the full plan was actually a bug, we just never noticed it because we never did manual selection inside MPyL (it was hacked in the pipeline before).

Reply:
We did use selected_plan before, just not in the plan creation. You changed a lot of things lately so stuff moved around, but I kept track of the usage and I don't think anything was broken at any point.

Reply:
I don't think that's correct, we never used `selected_plan` to write the JSON. This functionality seems to have been introduced [in this commit back in May](https://github.com/Vandebron/gh-mpyl/commit/c46d32292977ed0b2b93c4078f40b73e92b70e42#diff-5f8bc6ee933a7bacafed256143948c5c0b696a67bb5cddab95f2e0f3578f9dbaR54) and you can clearly see it using the `full_plan` already 🙂

Reply:
Because you didn't want to write the selected json back in those days, since it was being called on run (so not plan creation only) and the selected plan contained the stage. So you would write an incomplete plan that would be uploaded and re-used in the next stage in the gha and then cause issues.

^ without looking too much into the details since it's not relevant anymore anyway